src/Microsoft.FeatureManagement.AspNetCore/FeatureGateEndpointFilterExtensions.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.FeatureManagement.AspNetCore
+{
+    /// <summary>
+    /// Extension methods that provide feature management integration for ASP.NET Core endpoint building.
+    /// </summary>
+    public static class FeatureGateEndpointFilterExtensions
+    {
+        /// <summary>
+        /// Adds a feature flag filter to the endpoint that controls access based on feature state.
+        /// </summary>
+        /// <param name="builder">The endpoint convention builder.</param>
+        /// <param name="featureName">The name of the feature flag to evaluate.</param>
+        /// <returns>The endpoint convention builder for chaining.</returns>
+        /// <remarks>
+        /// This extension method enables feature flag control over endpoint access. When the feature is disabled,
+        /// requests to the endpoint will return a 404 Not Found response. The targeting context is obtained
+        /// from the ITargetingContextAccessor registered in the service collection.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// endpoints.MapGet("/api/feature", () => "Feature Enabled")
+        ///     .WithFeatureGate("MyFeature");
+        /// </code>
+        /// </example>
+        public static IEndpointConventionBuilder WithFeatureGate(this IEndpointConventionBuilder builder, string featureName)
+        {
+            return builder.AddEndpointFilter(new FeatureGateEndpointFilter(featureName));
+        }
```

Comment:
Similar to my first comment, I think using "feature" as name is better for consistency.